ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh

```diff
@@ -269,7 +269,10 @@ function suite() {
         echo "${TESTS}" | grep "${TEST_SKIPS}" || { exit_code=$?; echo 'Error: no tests were found matching the TEST_SKIPS regex:'; echo "$TEST_SKIPS"; return $exit_code; } &&
         TEST_ARGS="${TEST_ARGS:-} --file /tmp/tests"
     fi &&
-
+    # TO-DO remove heterogeneous block after manifest list support in internal registry
+    if [ -f "${SHARED_DIR}/heterogeneous_tests" ]; then
+        TEST_ARGS="${TEST_ARGS:-} --file ${SHARED_DIR}/heterogeneous_tests"
```

Comment:
I'm a bit confused about how this will interact with `TEST_SKIPS`.  Can we use `TEST_SKIPS` for the stopgap heterogeneous exclusions?  Or is it too hard to build the large skip list into a regexp or something?

Reply:
I tried handling the list into a regex but it was more complicated for a temporary effort. There aren't much common patterns between the list (70 tests)

Reply:
Another alternative Deep and I were discussing is like:
```bash 
TESTS="$(openshift-tests run --dry-run --provider "${TEST_PROVIDER}" "${TEST_SUITE}")" &&
if [[ -n "${TEST_SKIPS}" ]]; then
    TESTS=$(echo "${TESTS}" | grep -v "${TEST_SKIPS}") &&
    EXCLUDED_TESTS=$(echo "${TESTS}" | grep "${TEST_SKIPS}") &&
    [[ -n "${EXCLUDED_TESTS}" ]] || { exit_code=$?; echo 'Error: no tests were found matching the TEST_SKIPS regex:'; echo "$TEST_SKIPS"; return $exit_code; } &&
    echo "Skipping tests:\n${EXCLUDED_TESTS}"
fi &&
if [ -f "${SHARED_DIR}/excluded_tests" ]; then
    TESTS=$(echo $TESTS | grep -v -F -f ${SHARED_DIR}/excluded_tests) &&
    EXCLUDED_TESTS="${EXCLUDED_TESTS}\n$(<${SHARED_DIR}/excluded_tests)" &&
    echo "Skipping tests:\n${EXCLUDED_TESTS}"
fi &&
if [[ -n "${EXCLUDED_TESTS}" ]]; then
    echo ${TESTS} > /tmp/tests &&
    TEST_ARGS="${TEST_ARGS:-} --file /tmp/tests"
fi &&
```

Regardless of the temporary work related to the heterogeneous clusters in this PR, it would honor both TEST_SKIPS and the excluded_tests file and would work for any user needing this behavior.

Reply:
It will go away with IR manifest support for imagestream tags